protocol.go

```diff
@@ -153,8 +153,8 @@ func (p *Listener) Addr() net.Addr {
 func NewConn(conn net.Conn, opts ...func(*Conn)) *Conn {
 	// For v1 the header length is at most 108 bytes.
 	// For v2 the header length is at most 52 bytes plus the length of the TLVs.
-	// We use 256 bytes to be safe.
-	const bufSize = 256
+	// We use 65535, as it's the maximum length representable by a uint16.
+	const bufSize = 65535
```

Comment:
The spec seems to be adamant about the maximum sizes of both v1 and v2 headers, as declared in comments above this line, so I'm wondering what kind of headers are you parsing. Can you, please, implement a test w/ one or more examples of headers that go beyond 256 bytes?